controllers/image_generator.go

```diff
@@ -142,6 +142,11 @@ func newImageGenerator(client client.Client) (*ImageGenerator, error) {
 	}
 	ig.fips = fips == 1
 
+	err = ig.setupCloudProvider()
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup cloud provider: %v", err)
+	}
+
```

Comment:
Please provide a change log in this commit to explain why this needs to be done earlier.

Reply:
I'm not sure it's possible to change it, it shouldn't harm i think, however, I'd  be worried about messing the whole cluster

Reply:
Well... I tend to assume that anything can be changed in k8s by default and it is up to the operator to cope with that :wink: . Now if changing the infra object in our back has a greater impact than just breaking the operator, I guess it is acceptable.